src/sentry/search/events/builder/metrics.py

```diff
@@ -135,12 +135,17 @@ def _get_on_demand_metric_spec(self, field: str) -> Optional[OnDemandMetricSpec]
             if self.params.environments:
                 environment = self.params.environments[0].name
 
+            if not self.builder_config.on_demand_metrics_type:
+                raise InvalidSearchQuery(
+                    "Must include on demand metrics type when querying on demand"
+                )
+
             return OnDemandMetricSpec(
                 field=field,
                 query=self.query,
                 environment=environment,
                 groupbys=groupby_columns,
-                spec_type=self.builder_config.on_demand_metrics_type or MetricSpecType.SIMPLE_QUERY,
```

Comment:
This was originally excluded from the previous PR, somehow it got back in during a stash apply.